### PR TITLE
P4-2401 list unique journeys

### PR DESCRIPTION
**Pull request by jcavell**
Sorry, this is a big PR that now does 3 things:

* lists unique journeys that have a null price, ordered by null locations first
Screenshot 2020-11-11 at 23 03 41

* persist events during import

* get move by id (which returns a move and journey with events for these)
Example URL for this is http://localhost:8080/moves/04d75061-5bd4-4efa-8907-443812e00633

### CODE REVIEW

**Review by danielepolencic — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 1]
```diff
@@ -24,7 +26,7 @@ import javax.validation.Valid
 data class MonthsWidget(val currentMonth: Date, val nextMonth: Date, val previousMonth: Date)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:27]: Might be worth adding some minimal logging in the controller...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 2]
```diff
@@ -73,20 +64,38 @@ class HtmlController(@Autowired val movesForMonthService: MovesForMonthService)
         return "moves"
     }
 
+    @RequestMapping(JOURNEYS_URL)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:67]: For clarity probably worth changing to use the explicit `@GetMapping` and `@PostMapping` annotations in the controllers.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Review by danielepolencic — APPROVED** [review]

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**